### PR TITLE
TAF SHA-1 discussion

### PR DESCRIPTION
**Pull request by renatav**

### CODE REVIEW

File: POUFs/TAF-POUF/pouf2.md [hunk 1]
```diff
@@ -26,27 +26,61 @@ for metadata generation, it implicitly depends on the reference implementation
...
+it makes no claims about the integrity of their contents. TAF relies on Git's
+default artifact integrity protection. Git, however, still primarily relies on
+SHA-1, even though it has been proven that this hash function is vulnerable to
+collision attacks. If an attacker manages to gain access to the original target
+repositories, they could potentially exploit the SHA-1 weakness. As TAF stores
+verified URLs of its target repositories, users are protected from _mirrors_ of
+legitimate repositories presenting a colliding artifact in place of the
+original artifact.
+
+<!--
```
Comment (adityasaky) [inline POUFs/TAF-POUF/pouf2.md]: I think this may be overkill. I took a pass at focusing more on TAF + Git rather than SHAttered. WDYT? @renatav

Comment (adityasaky) [inline POUFs/TAF-POUF/pouf2.md:58]: I think this may be overkill. I took a pass at focusing more on TAF + Git rather than SHAttered. WDYT? @renatav

Comment (renatav) [reply POUFs/TAF-POUF/pouf2.md]: I am not sure, Justin suggested the following (quotes from our notes):

"I think you're better off just making the argument that this is a data store and data type that you need, here are the pros and cons. I think it's just better off leading with here's a usability but it's not the best for security. Our security is taking a substantial hit" 
So my understanding is that Justin wants us to be very clear that Git's security is not up to our standards and explain why we still want to use it for now

Comment (adityasaky) [reply POUFs/TAF-POUF/pouf2.md]: I think we do call out the weakness of SHA-1 above and that we directly use whatever Git uses. I'm wary of devoting too much space to SHAttered. @JustinCappos could you weigh in?

File: POUFs/TAF-POUF/pouf2.md [hunk 2]
```diff
@@ -26,27 +26,61 @@ for metadata generation, it implicitly depends on the reference implementation
...
+detected by TAF. TAF's users are encouraged to store signing keys for the
+`target` and `root` roles offline on hardware tokens to ensure their safety.
+
+In essence, while TAF ensures the validity of commits in target repositories,
+it makes no claims about the integrity of their contents. TAF relies on Git's
+default artifact integrity protection. Git, however, still primarily relies on
+SHA-1, even though it has been proven that this hash function is vulnerable to
+collision attacks. If an attacker manages to gain access to the original target
+repositories, they could potentially exploit the SHA-1 weakness. As TAF stores
+verified URLs of its target repositories, users are protected from _mirrors_ of
```
Comment (JustinCappos) [inline POUFs/TAF-POUF/pouf2.md]: What does verified URLs mean here?

File: POUFs/TAF-POUF/pouf2.md (hunk 1 above)
Comment (JustinCappos) [reply POUFs/TAF-POUF/pouf2.md]: I agree this paragraph should be cut.  We should be clear about the impact and leave it at that.

If I'm not being clear, let's hop on a call!

Comment (adityasaky) [reply POUFs/TAF-POUF/pouf2.md]: I removed the text about SHAttered.

**Review by JustinCappos — APPROVED** [review]

**Review by JustinCappos — APPROVED** [review]

**Comment by adityasaky** [comment]
Thanks! I've applied the commits into the target branch directly to avoid polluting the git history.

**Pull request closed without merging**